src/libCli/Completion.cpp

```diff
@@ -19,198 +19,252 @@ using namespace ArgParse;
+            {
+                std::string candidateStr = candidate->getMatchedStringRaw();
+                printf("pre: '%s'\n", candidateStr.c_str());
+            }
         }
-    }
 
-    size_t maxSuggestionLen = 0;
-    size_t maxSuggestionDocLen = 0;
+        size_t maxSuggestionLen = 0;
+        size_t maxSuggestionDocLen = 0;
 
-    // list of suggestions. first=completed string, second=docstring
-    std::vector<Suggestion> suggestions;
-    size_t n = f_args.size();
+        // list of suggestions. first=completed string, second=docstring
+        std::vector<Suggestion> suggestions;
+        size_t n = f_args.size();
 
-    for(auto candidate : f_candidates)
-    {
-        std::string candidateStr =candidate->getMatchedStringRaw();
-        Suggestion suggestion;
-        size_t start = n;
-        size_t end;
-        if(f_debug)
-        {
-            printf("candidateStr[n=%zu] = '%c'\n", n, candidateStr[n]);
-        }
-        if(
-                (candidateStr[n] != ' ')
-                &&
-                (candidateStr[n] != '=')
-                &&
-                (candidateStr[n] != ',')
-                &&
-                (candidateStr[n] != ':')
-        )
+        for (auto candidate : f_candidates)
         {
-            // bash always expects completion suggestions to start from the last token.
-            // Now we need to find out where the last token has started
-            // We need to "simulate" bash tokenizer here. tokens are delimited by ' ' '=' or ':'
-            start = candidateStr.find_last_of(" =:,", n)+1;
-            if(start == std::string::npos)
+            std::string candidateStr = candidate->getMatchedStringRaw();
+            Suggestion suggestion;
+            size_t start = n;
+            size_t end;
+            if (f_debug)
             {
-                start = 0;
+                printf("candidateStr[n=%zu] = '%c'\n", n, candidateStr[n]);
+            }
+            if (
+                (candidateStr[n] != ' ') &&
+                (candidateStr[n] != '=') &&
+                (candidateStr[n] != ',') &&
+                (candidateStr[n] != ':'))
+            {
+                // bash always expects completion suggestions to start from the last token.
+                // Now we need to find out where the last token has started
+                // We need to "simulate" bash tokenizer here. tokens are delimited by ' ' '=' or ':'
+                start = candidateStr.find_last_of(" =:,", n) + 1;
+                if (start == std::string::npos)
+                {
+                    start = 0;
+                }
+                end = candidateStr.find_first_of(" ", n) - 1;
+                //printf("cand='%s', n=%zu, start=%zu, end = %zu\n",candidateStr.c_str(), n, start, end);
+            }
+            else
+            {
+                start = candidateStr.find_last_of(" =:", n - 1) + 1;
+                //end = candidateStr.find_first_of(' ', n+1);
```

Comment:
dead code

Reply:
This one is for the bash-completion. Since I did not write this code, I don't know what will happen if this is removed.